spec/app/domain/authentication/authn-jwt/signing_key/fetch_public_keys_signing_key_spec.rb

```diff
@@ -0,0 +1,102 @@
+      account: account,
+      username: "dummy_identity",
+      credentials: "dummy",
+      client_ip: "dummy",
+      request: "dummy"
+    )
+  }
+
+  let(:required_jwks_uri_configuration_error) { "required jwks_uri configuration missing error" }
+  let(:bad_response_error) { "bad response error" }
+  let(:required_secret_missing_error) { "required secret missing error" }
+  let(:mocked_logger) { double("Mocked Logger")  }
+  let(:mocked_fetch_signing_key) { double("MockedFetchSigningKey") }
+  let(:mocked_fetch_signing_key_refresh_value) { double("MockedFetchSigningKeyRefreshValue") }
+  let(:mocked_fetch_authenticator_secrets_exist_values)  {  double("MockedFetchAuthenticatorSecrets") }
+  let(:mocked_fetch_authenticator_secrets_empty_values)  {  double("MockedFetchAuthenticatorSecrets") }
+  let(:mocked_bad_http_response) { double("Mocked bad http response")  }
+  let(:mocked_good_http_response) { double("Mocked good http response")  }
+  let(:mocked_bad_response) { double("Mocked bad http body")  }
+  let(:mocked_good_response) { double("Mocked good http body")  }
+  let(:mocked_create_jwks_from_http_response) { double("Mocked good jwks")  }
+
+  let(:good_response) { "good-response"}
+  let(:bad_response) { "bad-response"}
+  let(:valid_jwks) { "valid-jwls" }
+
+  before(:each) do
+    allow(mocked_logger).to(
+      receive(:call).and_return(true)
+    )
+
+    allow(mocked_logger).to(
+      receive(:debug).and_return(true)
+    )
+
+    allow(mocked_logger).to(
+      receive(:info).and_return(true)
+    )
+
+    allow(mocked_fetch_signing_key).to receive(:call) { |params| params[:signing_key_provider].fetch_signing_key }
+    allow(mocked_fetch_signing_key_refresh_value).to receive(:call) { |params| params[:refresh] }
+
+    allow(mocked_fetch_authenticator_secrets_exist_values).to(
+      receive(:call).and_return('jwks-uri' => 'https://jwks-uri.com/jwks')
+    )
+
+    allow(mocked_fetch_authenticator_secrets_empty_values).to(
+      receive(:call).and_raise(required_secret_missing_error)
+    )
+
+    allow(mocked_bad_http_response).to(
+      receive(:get_response).and_return(bad_response)
+    )
+
+    allow(mocked_good_http_response).to(
+      receive(:get_response).and_return(good_response)
+    )
+
+    allow(mocked_create_jwks_from_http_response).to(
+      receive(:call).with(http_response: good_response).and_return(valid_jwks)
+    )
+
+    allow(mocked_create_jwks_from_http_response).to(
+      receive(:call).with(http_response: bad_response).and_raise(bad_response_error)
+    )
+  end
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
+  context "FetchPublicKeysSigningKey call " do
+    context "propagates false refresh value" do
+      subject do
+        jwks = Net::HTTP.get_response(URI("https://www.googleapis.com/oauth2/v3/certs")).body
```

Comment:
It means that our automation will be dependent on the existence of this endpoint? 
Its not better this thing be in integration test and with our own endpoint for it?

Reply:
I guess that there're billions things depends this endpoint in the real world :-).
Let's consider UTs as not finished yet but I'm almost sure that I'll keep it because it will be a great indicator of Conjur's compatibility to the real world...